karaf-assembly/pentaho-karaf-assembly-client/pom.xml

```diff
@@ -0,0 +1,259 @@
+      <classifier>features</classifier>
+      <version>${project.version}</version>
+      <type>xml</type>
+      <scope>runtime</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.pentaho.di.plugins</groupId>
+      <artifactId>get-fields-plugin</artifactId>
+      <classifier>features</classifier>
+      <version>${project.version}</version>
+      <type>xml</type>
+      <scope>runtime</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.pentaho</groupId>
+      <artifactId>pentaho-requirejs-osgi-manager</artifactId>
+      <classifier>features</classifier>
+      <version>${project.version}</version>
+      <type>xml</type>
+      <scope>runtime</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.pentaho.webpackage</groupId>
+      <artifactId>pentaho-webpackage</artifactId>
+      <classifier>features</classifier>
+      <version>${project.version}</version>
+      <type>xml</type>
+      <scope>runtime</scope>
+    </dependency>
+    <dependency>
+      <groupId>pentaho</groupId>
+      <artifactId>pentaho-dataservice</artifactId>
+      <classifier>features</classifier>
+      <version>${project.version}</version>
+      <type>xml</type>
+      <scope>runtime</scope>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <version>${maven-dependency-plugin.version}</version>
+        <executions>
+          <execution>
+            <id>unpack</id>
+            <phase>generate-resources</phase>
+            <goals>
+              <goal>unpack</goal>
+            </goals>
+            <configuration>
+              <artifactItems>
+                <artifactItem>
+                  <groupId>org.pentaho</groupId>
+                  <artifactId>common-ui-impl-client-config</artifactId>
+                  <version>${project.version}</version>
+                  <classifier>configuration-bundle</classifier>
+                  <outputDirectory>target/classes/config/web-client</outputDirectory>
+                </artifactItem>
+              </artifactItems>
+              <excludes>/META-INF/**</excludes>
+            </configuration>
+          </execution>
+        </executions>
+        <inherited>false</inherited>
+      </plugin>
+
+      <!-- karaf-maven-plugin will call both assembly and archive goals -->
+      <plugin>
+        <groupId>org.hitachivantara.karaf.tooling</groupId>
+        <artifactId>karaf-maven-plugin</artifactId>
+        <version>${karaf-maven-plugin.version}</version>
+        <extensions>true</extensions>
+        <configuration>
+          <archiveTarGz>false</archiveTarGz>
+          <archiveZip>true</archiveZip>
+          <!-- override expected osgi.ee to match compilation target -->
+          <javase>1.8</javase>
+          <!-- install startup jars 'by-reference' which avoids copying -->
+          <useReferenceUrls>false</useReferenceUrls>
```

Comment:
Why is this being used? From the code it seems that it has nothing to do with the comment.
https://github.com/apache/karaf/blob/karaf-4.2.6/tooling/karaf-maven-plugin/src/main/java/org/apache/karaf/tooling/AssemblyMojo.java#L339-L347

Reply:
useReferenceUrls and ignoreDependencyFlag are already false by default